src/content/app/genome-browser/state/browser-general/browserGeneralSlice.ts

```diff
@@ -0,0 +1,215 @@
+  const state = getState();
+  const activeEnsObjectIdsMap = getBrowserActiveEnsObjectIds(state);
+  Object.values(activeEnsObjectIdsMap).forEach((objectId) =>
+    dispatch(fetchEnsObject(objectId))
+  );
+};
+
+export const updateTrackStatesAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: BrowserTrackStates) => (dispatch, getState: () => RootState) => {
+  dispatch(updateTrackStates(payload));
+  const trackStates = getBrowserTrackStates(getState());
+  browserStorageService.saveTrackStates(trackStates);
+};
+
+export const setDataFromUrlAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: ParsedUrlPayload) => (dispatch) => {
+  dispatch(setEntityDataFromUrl(payload));
+  dispatch(setLocationDataFromUrl(payload));
+  dispatch(setInitialTrackPanelDataForGenome(payload));
+
+  const { activeGenomeId, activeEnsObjectId, chrLocation } = payload;
+
+  dispatch(ensureSpeciesIsEnabled(payload.activeGenomeId));
+  browserStorageService.saveActiveGenomeId(payload.activeGenomeId);
+  chrLocation &&
+    browserStorageService.updateChrLocation({ [activeGenomeId]: chrLocation });
+
+  if (activeEnsObjectId) {
+    browserStorageService.updateActiveEnsObjectIds({
+      [activeGenomeId]: activeEnsObjectId
+    });
+  }
+};
+
+export const updateBrowserActiveEnsObjectIdsAndSave = (
+  activeEnsObjectId: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (!activeGenomeId) {
+      return;
+    }
+    const currentActiveEnsObjectIds = getBrowserActiveEnsObjectIds(state);
+    const updatedActiveEnsObjectIds = {
+      ...currentActiveEnsObjectIds,
+      [activeGenomeId]: activeEnsObjectId
+    };
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+    dispatch(fetchEnsObject(activeEnsObjectId));
+
+    browserStorageService.updateActiveEnsObjectIds(updatedActiveEnsObjectIds);
+  };
+};
+
+export const deleteSpeciesInGenomeBrowser = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+
+    dispatch(deleteGenomeEntityData(genomeIdToRemove));
+    dispatch(deleteGenomeLocationData(genomeIdToRemove));
+    dispatch(deleteGenomeTrackPanelData(genomeIdToRemove));
+
+    const updatedActiveEnsObjectIds = pickBy(
+      getBrowserActiveEnsObjectIds(state),
+      (value, key) => key !== genomeIdToRemove
+    );
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+
+    browserStorageService.deleteGenome(genomeIdToRemove);
+    trackPanelStorageService.deleteGenome(genomeIdToRemove);
+  };
+};
+
+const browserGeneralSlice = createSlice({
+  name: 'genome-browser-entity',
+  initialState: defaultBrowserEntityState,
```

Comment:
This line and the previous line still have the words "browser entity" in them. There may be more. Could you search through this file and check? There are also other usages of the word "entity" in this file, not just browser entity; so it can't just be automatically replaced in one go.

Reply:
That's  done